src/LoginPage.cpp

```diff
@@ -186,13 +183,26 @@ LoginPage::loginError(const QString &msg)
         error_label_->setText(msg);
 }
 
+bool
+LoginPage::isMatrixIdValid()
+{
+        QRegularExpressionValidator v(QRegularExpression("@.+?:.{3,}"), this);
+        QString s = matrixid_input_->text();
+        int pos   = 0;
+        return v.validate(s, pos);
+}
+
 void
 LoginPage::onMatrixIdEntered()
 {
         error_label_->setText("");
 
         User user;
 
+        if (isMatrixIdValid() == 0) {
+                return loginError("You have entered an invalid Matrix ID  e.g @joe:matrix.org");
```

Comment:
```suggestion
        if (!isMatrixIdValid()) {
                loginError("You have entered an invalid Matrix ID  e.g @joe:matrix.org");
                return;
```

Using return with functions returning void is unusual, so I prefer to use 2 statements instead. Booleans should be compared to true or false or be used directly.